tripal/includes/TripalEntityUIController.inc

```diff
@@ -115,7 +115,20 @@ class TripalEntityUIController extends EntityDefaultUIController {
  * @see hook_entity_view_alter()
  */
 function tripal_view_entity($entity, $view_mode = 'full') {
-   $content = '';
+   if(!is_object($entity)) {
+     $id = intval($entity);
+     if($id === 0) {
+       return drupal_not_found();
```

Comment:
Should we return `drupal_not_found()` here or simply return NULL?

Reply:
This seems reasonable to me.